scripts/clear-signed-agreements.py

```diff
@@ -41,13 +47,22 @@ def main(stage, framework_slug, api_token, user):
     )
 
     suppliers = api_client.find_framework_suppliers(framework_slug, agreement_returned=True)['supplierFrameworks']
-    for supplier in suppliers:
+
+    if supplier_ids is not None:
+        supplier_ids = [
+            str(supplier['supplierId']) for supplier in suppliers
+            if str(supplier['supplierId']) in supplier_ids.split(',')
+        ]
+    else:
+        supplier_ids = [str(supplier['supplierId']) for supplier in suppliers]
+
+    for supplier_id in supplier_ids:
         logger.info("Resetting agreement returned flag for supplier {supplier_id}",
-                    extra={'supplier_id': supplier['supplierId']})
-        api_client.unset_framework_agreement_returned(supplier['supplierId'], framework_slug, user)
+                    extra={'supplier_id': supplier_id})
+        api_client.unset_framework_agreement_returned(supplier_id, framework_slug, user)
 
     signed_agreements = filter(
-        lambda x: re.search(r'/(\d+)-signed-framework-agreement.pdf', x['path']),
+        lambda x: x['path'] in ['{}-signed-framework-agreement.pdf'.format(id) for id in supplier_ids],
```

Comment:
We should remove `.pdf`, since uploaded agreements could be scanned images of last page. I had to fix it manually last time, but forgot to commit.